src/DIRAC/Resources/Computing/AREXComputingElement.py

```diff
@@ -0,0 +1,648 @@
+            )
+            dID = ""
+            if r.ok:  # Get the delegation and "PUT" it in the CE ...
+                dID = r.headers.get("location", "")
+                if len(dID) > 2:
+                    dID = dID.split("new/")[-1]
+                    command = "delegations/" + dID
+                    query = self.base_url + command
+                    r1 = self.s.put(query, data=r.text, headers=self.headers, timeout=self.arcRESTTimeout)
+                    if not r1.ok:
+                        dID = ""
+                else:
+                    dID = ""
+            return dID
+        else:  # Retrieve delegation for existing job
+            jj = {"job": [{"id": jobID}]}  # job in ARC REST json format
+            command = "jobs"
+            params = {"action": "delegations"}
+            query = self.base_url + command
+            r = self.s.post(query, data=json.dumps(jj), headers=self.headers, timeout=self.arcRESTTimeout)
+            dID = ""
+            if r.ok:  # Check if the job has a delegation
+                p = json.loads(r.text.replace("\n", ""))
+                if "delegation_id" in p["job"]:
+                    dID = p["job"]["delegation_id"][0]
+            return dID
+
+    #############################################################################
+    def submitJob(self, executableFile, proxy, numberOfJobs=1):
+        """Method to submit job
+        Assume that the ARC queues are always of the format nordugrid-<batchSystem>-<queue>
+        And none of our supported batch systems have a "-" in their name
+        """
+        if not self.initialised:  # Something went wrong in the initialisation. Redo it.
+            return S_ERROR("REST interface not initialised. Cannot submit jobs.")
+
+        self.arcQueue = self.queue.split("-", 2)[2]
+
+        result = self._prepareProxy()
+        if not result["OK"]:
+            self.log.error("AREXComputingElement: failed to set up proxy", result["Message"])
+            return result
+        self.s.cert = Locations.getProxyLocation()
+
+        self.log.verbose("Executable file path: %s" % executableFile)
+        if not os.access(executableFile, 5):
+            os.chmod(executableFile, stat.S_IRWXU | stat.S_IRGRP | stat.S_IXGRP | stat.S_IROTH + stat.S_IXOTH)
+
+        batchIDList = []
+        stampDict = {}
+
+        command = "jobs"
+        params = {"action": "new"}  # Is this the recommended way?
+        query = self.base_url + command
+        # First get a "delegation" and use the same delegation for all the jobs
+        deleg = self._delegation("")
+        deleText = ""
+        if len(deleg) < 2:
+            self.log.warning("Could not get a delegation", "For CE %s", self.ceHost)
+            self.log.warning("Continue without a delegation")
+        else:
+            deleText = "(delegationid=%s)" % deleg
+        # It is fairly simple to change to bulk submission. Should I do so?
```

Comment:
As an explanation : XRSL uses "+" to merge jobs. So, if we want to submit multiple jobs, we need a long string along the lines of

+(&(...))(&(...))(&(...))

where each "&(...)" is a pilot job for us.

I do not know how big each job in DIRAC is in terms of the length of the string used. As a result, if we put together too many jobs for multiple submission, I worry the following two points may affect the submission 

1. Internal string length within python
2. How much "data" can the requests module of python support (https://docs.python-requests.org/en/master/index.html)?

The above questions are for experts in the field to answer. I would naively guess that they are not an issue, but wanted to get a basic version into production without having to debug this as a possible problem.

If neither of the above is a problem, we can happily use the above prescription to conflate multiple jobs and submit them in one interaction with the CE.

Reply:
I don't think your worries are an issue, so you can conflate multiple jobs in one go.

Reply:
Okay - thanks. Once I am able to submit to the certification system again, I will test this and implement it.

Reply:
> Internal string length within python

However much RAM you have

> How much "data" can the requests module of python support (https://docs.python-requests.org/en/master/index.html)?

IIRC there is a 2GB limit here but if the payload size is anywhere near that big then we'll hit other issues.

One thing to be careful of is that `xrslString = xrslString + "something"` can cause performance issues when the string gets long as you have to copy the string every time and the runtime becomes quadratic. It's better to use something like this:

```python
xrslStrings = []
for job in jobs:
    xrslString = ...
    xrslStrings.append(xrslString)
xrslString = "&".join(xrslString)
```

Reply:
Thanks. Will keep in mind for when I am able to test on the certification instance. I really have no feel about the size of the payload, but hope that it is much less than 2GB - unless of course we are doing something mad.

As a comment though, I suppose in our case it will have to simply be n copies of the same string added together. How would you suggest I do this efficiently? Will something like the following work (note the structure in https://github.com/DIRACGrid/DIRAC/pull/5542#discussion_r744189927)?

```
multipleJobs = "+(" + ")(".join(n*[xrslString]) + ")"
```
Where n is of course the number of jobs I want to submit.

Reply:
It turns out that there is yet another bug in ARC which prevented multiple job submission. This is also being fixed (hopefully) in the next version.

I have also had a comment from the developer who does not think that submitting multiple jobs would be faster than our iterative submission of single jobs. I have asked for clarification on why.

Reply:
And from response of the ARC developer as to "why will this *not* be faster?"
-----
It will take same amount (actually slightly more due to additional parsing of catenated job descriptions) of CPU power to process job descriptions. You can save on request/response time in case of low latency network. But as nothing prevents client form sending multiple HTTP POSTs without waiting for answer I expect the difference to be negligible. Of course it may vary due too many variables.
-----
Full thread in nordugrid mailing list in https://mail.nordugrid.org/mailman/private/nordugrid-discuss/2021q4/055051.html
-----
So, there is a question as to whether we want to do this, and how much of a benefit will DIRAC get from multiple job submission.

Reply:
Right. So, don't bother with bulk.